Loop/Managers/TestingScenariosManager.swift

```diff
@@ -233,9 +233,12 @@ extension TestingScenariosManagerRequirements {
             }
         }
         
-        instance.injectedActions.forEach { [testingCGMManager, testingPumpManager] action in
-            testingCGMManager?.inject(action: action)
-            testingPumpManager?.inject(action: action)
+        instance.deviceActions.forEach { [testingCGMManager, testingPumpManager] action in
+            if testingCGMManager?.managerIdentifier == action.managerIdentifier {
```

Comment:
I also wonder if a user tries to load a scenario and they don't have the right devices configured, if we should alert them.